File: api/src/main/java/net/blueshell/api/mapper/EventSocialMapper.java
package net.blueshell.api.mapper;

import net.blueshell.api.common.enums.PlatformType;
import net.blueshell.api.dto.EventDTO;
import net.blueshell.api.dto.SocialDTO;
import org.mapstruct.AfterMapping;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public abstract class EventSocialMapper {

    @Mapping(source = "description", target = "text")
    @Mapping(target = "url", ignore = true)
    @Mapping(target = "platforms", ignore = true)
    public abstract SocialDTO toSocialDto(EventDTO dto);


    @AfterMapping
    public void afterToSocialDTO(EventDTO dto, @MappingTarget SocialDTO socialDTO) {
        PlatformType[] platforms = {PlatformType.FACEBOOK, PlatformType.TWITTER, PlatformType.INSTAGRAM};
        socialDTO.setPlatforms(platforms);
    }
}

File: api/src/main/kotlin/net/blueshell/api/feature/event/mapper/EventSocialMapper.kt
package net.blueshell.api.feature.event.mapper

import net.blueshell.api.shared.enums.PlatformType
import net.blueshell.api.feature.blog.dto.SocialDTO
import net.blueshell.api.feature.event.dto.EventDTO
import org.mapstruct.*

@Mapper(componentModel = "spring")
abstract class EventSocialMapper {
    @Mapping(source = "description", target = "text")
    @BeanMapping(ignoreByDefault = true)
    abstract fun toSocialDto(dto: EventDTO): SocialDTO

    @AfterMapping
    fun afterToSocialDTO(dto: EventDTO, @MappingTarget socialDTO: SocialDTO) {
        val platforms = arrayOf(PlatformType.FACEBOOK, PlatformType.TWITTER, PlatformType.INSTAGRAM)
        socialDTO.platforms = platforms
    }
}
